pages/common/nnn.md

```diff
@@ -0,0 +1,36 @@
+# nnn
+
+> Interactive terminal file manager and disk usage nalyzer.
+> More information: <https://github.com/jarun/nnn>.
+
+- Open the current directory (or specify one as the first argument):
+
+`nnn`
+
+- Open an existing bookmark:
+
+`nnn -b {{bookmark_name}}`
+
+- Show hidden files:
+
+`nnn -d`
+
+- Start in disk usage analyzer mode:
+
+`nnn -S`
+
+- Start in light mode (fewer details):
+
+`nnn -l`
+
+- Copy selected file path to file_out, or stdout if file_out='-':
+
+`nnn -p {{file_out}}`
+
+- Use substring matching for filters instead of regex:
+
+`nnn -s`
+
+- Use version compare to sort files:
```

Comment:
This is not clear to me. From where is it getting the version ?

Reply:
Good point, at a quick look from building the command I thought it was meant for ordering files based on the version string in their names, for example `abc-1.10` vs `abc-1.2`, but _it does not look like so_ (order should be `1.1`, `1.2`, `1.10` in such case):

    $ touch abc-1.1
    $ touch abc-1.2
    $ touch abc-1.10
    $ nnn

    2019-06-07 12:43 644      0B abc-1.1
    2019-06-07 12:43 644      0B abc-1.10
    2019-06-07 12:43 644      0B abc-1.2

    $ nnn -s

    2019-06-07 12:43 644      0B abc-1.1
    2019-06-07 12:43 644      0B abc-1.10
    2019-06-07 12:43 644      0B abc-1.2

Some explanation would be helpful indeed :thinking:

Reply:
I don't know. We need clarity on how the sorting is done in the first place. If someone can explain that, that would be great.

Reply:
To me, that looks like a string-based comparison sort - i.e. it compares each character one at a time. On the flip-side, a _natural_ sort would look like this:

```
1.1
1.2
1.10
```

This uses the terminology used in the PHP documentation.

Reply:
@sbrl yes, we came to the same conclusion. What we were trying to understand was what actually is the difference between `nnn` and `nnn -s`...

Reply:
Ah, I see. Is there a manual that we can refer to at all, or a developer we can @mention and ask?

Reply:
This is the GH project, https://github.com/jarun/nnn 

Maybe @jarun could be so kind to help us figure this out?

Reply:
Hey guys,

Thanks for considering `nnn`. Please take a look at the program help from master:

```
usage: nnn [-a] [-b key] [-d] [-f] [-H] [-i] [-n] [-o]
           [-p file] [-r] [-s] [-S] [-t] [-v] [-h] [PATH]

The missing terminal file manager for X.

positional args:
  PATH   start dir [default: current dir]

optional args:
 -a      use access time
 -b key  open bookmark key
 -d      detail mode
 -f      run filter as cmd on prompt key
 -H      show hidden files
 -i      nav-as-you-type mode
 -n      version sort
 -o      press Enter to open files
 -p file selection file (stdout if '-')
 -r      show cp, mv progress on Linux
 -s      string filters [default: regex]
 -S      du mode
 -t      disable dir auto-select
 -v      show version
 -h      show help
```

I am suggesting master because there would be some changes in the upcoming release.

The switch to version sort is `-n`. The option `-s` is to use plain text strings instead of regex at the filter prompt.